torch/distributed/fsdp/_optim_utils.py

```diff
@@ -634,11 +634,17 @@ def _flatten_optim_state(
             )
             # Shard the flattened tensor immediately to minimize max memory
             # usage
-            sharded_flat_tensor, _ = FlatParamHandle._get_shard(
-                flat_tensor,
-                fsdp_state.rank,
-                fsdp_state.world_size,
-            )
+            if (
+                fsdp_state.world_size == 1
```

Comment:
Mind adding some comment around this check?
e.g. what this `== 1` check is for and why is it not `!= 1`?

Reply:
Correct, this was a mistake.